logisland-components/logisland-services/logisland-service-elasticsearch/logisland-service-elasticsearch_7_x-client/src/integration-test/java/com/hurence/logisland/service/elasticsearch/Elasticsearch_7_x_ClientServiceIT.java

```diff
@@ -0,0 +1,559 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hurence.logisland.service.elasticsearch;
+
+import com.hurence.logisland.classloading.PluginProxy;
+import com.hurence.logisland.component.InitializationException;
+import com.hurence.logisland.component.PropertyDescriptor;
+import com.hurence.logisland.controller.ControllerServiceInitializationContext;
+import com.hurence.logisland.processor.ProcessException;
+import com.hurence.logisland.record.FieldType;
+import com.hurence.logisland.record.Record;
+import com.hurence.logisland.record.StandardRecord;
+import com.hurence.logisland.service.datastore.DatastoreClientServiceException;
+import com.hurence.logisland.service.datastore.InvalidMultiGetQueryRecordException;
+import com.hurence.logisland.service.datastore.MultiGetQueryRecord;
+import com.hurence.logisland.service.datastore.MultiGetResponseRecord;
+import com.hurence.logisland.util.runner.TestRunner;
+import com.hurence.logisland.util.runner.TestRunners;
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.bulk.BulkItemResponse;
+import org.elasticsearch.action.bulk.BulkProcessor;
+import org.elasticsearch.action.bulk.BulkRequest;
+import org.elasticsearch.action.bulk.BulkResponse;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.common.unit.ByteSizeUnit;
+import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.common.unit.TimeValue;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.*;
+import java.util.function.BiConsumer;
+
+public class Elasticsearch_7_x_ClientServiceIT {
+
+    private static final String MAPPING1 = "{'properties':{'name':{'type': 'text'},'val':{'type':'integer'}}}";
+    private static final String MAPPING2 = "{'properties':{'name':{'type': 'text'},'val':{'type': 'text'}}}";
+    private static final String MAPPING3 =
+            "{'dynamic':'strict','properties':{'name':{'type': 'text'},'xyz':{'type': 'text'}}}";
+
+    private static Logger logger = LoggerFactory.getLogger(Elasticsearch_7_x_ClientServiceIT.class);
+
+    @Rule
+    public final ESRule esRule = new ESRule();
+
+    private class MockElasticsearchClientService extends Elasticsearch_7_x_ClientService {
+
+        @Override
+        protected void createElasticsearchClient(ControllerServiceInitializationContext context) throws ProcessException {
+            if (esClient != null) {
+                return;
+            }
+            esClient = esRule.getClient();
+        }
+
+        @Override
+        protected void createBulkProcessor(ControllerServiceInitializationContext context) {
```

Comment:
If I am correct, this method is the same than super so why are you overriding ?
This should be accomplished thourgh service properties in my opinion during configuration of service in 
configureElasticsearchClientService method line 134 of this file. So your test would also ensure that configuration is properly handled